libsql-sqlite3/src/wal.c

```diff
@@ -4250,150 +4151,210 @@ static void libsqlGetWalPathname(char *buf, const char *orig, int orig_len) {
+  int main_db_name_len = sqlite3Strlen30(main_db_path_name);
+
+  if( main_db_name_len > 0 ){
+      char *ptr = (char*)sqlite3MallocZero(libsqlWalPathnameLen(main_db_name_len) + 1);
+      if (!ptr) return SQLITE_NOMEM_BKPT;
+      libsqlGetWalPathname(ptr, main_db_path_name, main_db_name_len);
+      *out = ptr;
+  }
+
   return SQLITE_OK;
 }
 
-libsql_wal_methods *libsql_wal_methods_find(const char *zName) {
-  static libsql_wal_methods methods;
-  static libsql_wal_methods *methods_head = NULL; 
-
-#ifndef SQLITE_OMIT_AUTOINIT
-  int rc = sqlite3_initialize();
-  if (rc != SQLITE_OK) {
-    return NULL;
-  }
-#endif
+int sqlite3LogExists(void* self, sqlite3_vfs *vfs, const char *main_db_path_name, int *exists) {
+    char *zWal;
+    int rc = libsqlMakeWalPathname(main_db_path_name, &zWal);
+    if (rc != 0) return rc;
+    rc = sqlite3OsAccess(vfs, zWal, SQLITE_ACCESS_EXISTS, exists);
+    sqlite3_free(zWal);
+    if (rc != 0) return rc;
+    return SQLITE_OK;
+}
 
-  if (!zName || *zName == '\0') {
-    zName = "default";
-  }
-  if (methods_head == NULL && strncmp(zName, "default", 7) == 0) {
-    methods.iVersion = 1;
-    methods.xOpen = sqlite3WalOpen;
-    methods.xClose = sqlite3WalClose;
-    methods.xLimit = sqlite3WalLimit;
-    methods.xBeginReadTransaction = sqlite3WalBeginReadTransaction;
-    methods.xEndReadTransaction = sqlite3WalEndReadTransaction;
-    methods.xFindFrame = sqlite3WalFindFrame;
-    methods.xReadFrame = sqlite3WalReadFrame;
-    methods.xDbsize = sqlite3WalDbsize;
-    methods.xBeginWriteTransaction = sqlite3WalBeginWriteTransaction;
-    methods.xEndWriteTransaction = sqlite3WalEndWriteTransaction;
-    methods.xUndo = sqlite3WalUndo;
-    methods.xSavepoint = sqlite3WalSavepoint;
-    methods.xSavepointUndo = sqlite3WalSavepointUndo;
-    methods.xFrames = sqlite3WalFrames;
-    methods.xCheckpoint = sqlite3WalCheckpoint;
-    methods.xCallback = sqlite3WalCallback;
-    methods.xExclusiveMode = sqlite3WalExclusiveMode;
-    methods.xHeapMemory = sqlite3WalHeapMemory;
+int sqlite3LogDestroy(void* self, sqlite3_vfs *vfs, const char *main_db_path_name) {
+    char *zWal;
+    int rc = libsqlMakeWalPathname(main_db_path_name, &zWal);
+    if (rc != 0) return rc;
+    rc = sqlite3OsDelete(vfs, zWal, 0);
+    sqlite3_free(zWal);
+    if (rc != 0) return rc;
+    return SQLITE_OK;
+}
 
-#ifdef SQLITE_ENABLE_SNAPSHOT
-    methods.xSnapshotGet = sqlite3WalSnapshotGet;
-    methods.xSnapshotOpen = sqlite3WalSnapshotOpen;
-    methods.xSnapshotRecover = sqlite3WalSnapshotRecover;
-    methods.xSnapshotCheck = sqlite3WalSnapshotCheck;
-    methods.xSnapshotUnlock = sqlite3WalSnapshotUnlock;
-#endif
+/*
+** Open a connection to the WAL file zWalName. The database file must
+** already be opened on connection pDbFd. The buffer that zWalName points
+** to must remain valid for the lifetime of the returned Wal* handle.
+**
+** Custom virtual methods may be provided via the pMethods parameter.
```

Comment:
this comment is out of date right?

Reply:
yes yes, I haven't started looking at documentation, just on getting something in a working state